packages/contracts/src/contracts/current/protocol/Exchange/LibErrors.sol

```diff
@@ -21,13 +21,17 @@ pragma experimental ABIEncoderV2;
 
 contract LibErrors {
 
-    // Error Codes
+    // Status Codes
     enum Errors {
-        ORDER_EXPIRED,                    // Order has already expired
-        ORDER_FULLY_FILLED,               // Order has already been fully filled
-        ORDER_CANCELLED,                  // Order has already been cancelled
-        ROUNDING_ERROR_TOO_LARGE,         // Rounding error too large
-        INSUFFICIENT_BALANCE_OR_ALLOWANCE // Insufficient balance or allowance for token transfer
+        INVALID,                           // The first (default) value is invalid
+        SUCCESS,                           // Operation executed normaly
+        ORDER_INVALID,                     // Order is invalid
+        ORDER_SIGNATURE_INVALID,           // Signature invalid
+        ORDER_EXPIRED,                     // Order has already expired
+        ORDER_FULLY_FILLED,                // Order has already been fully filled
+        ORDER_CANCELLED,                   // Order has already been cancelled
+        INVALID_TAKER,                     // Order can not be filled by taker
```

Comment:
This value is not used.

Reply:
We were debating returning an error for invalid taker vs reverting. IMO, since the taker can always be validated off-chain before submitting the tx, we should keep the revert (this is no longer true for invalid signatures with some of the new types).